src/platform/packages/shared/response-ops/alerts-table/components/alerts_table.tsx

```diff
@@ -211,39 +211,52 @@ const AlertsTableContent = typedForwardRef(
     const { casesConfiguration, showInspectButton } = publicDataGridProps;
     const { data, cases: casesService, http, notifications, application, licensing } = services;
     const queryClient = useQueryClient({ context: AlertsQueryContext });
-    const storageRef = useRef(configurationStorage);
     const dataGridRef = useRef<EuiDataGridRefProps>(null);
-    const localStorageAlertsTableConfig = storageRef.current.get(
-      id
-    ) as Partial<AlertsTablePersistedConfiguration>;
 
-    const columnsLocal = useMemo(
-      () =>
-        !isEmpty(localStorageAlertsTableConfig?.columns)
-          ? localStorageAlertsTableConfig!.columns!
-          : !isEmpty(initialColumns)
-          ? initialColumns!
-          : [],
-      [initialColumns, localStorageAlertsTableConfig]
-    );
+    const [configuration, setConfiguration] = useAlertsTableConfiguration({
+      id,
+      configurationStorage,
+      notifications,
+    });
 
-    const getStorageConfig = useCallback(
-      () => ({
-        columns: columnsLocal,
-        sort: !isEmpty(localStorageAlertsTableConfig?.sort)
-          ? localStorageAlertsTableConfig!.sort!
-          : initialSort ?? [],
-        visibleColumns: !isEmpty(localStorageAlertsTableConfig?.visibleColumns)
-          ? localStorageAlertsTableConfig!.visibleColumns!
-          : columnsLocal.map((c) => c.id),
+    // Keeping a stable reference to the default columns to support the reset functionality and
+    // to apply default properties to the configured columns
+    const [defaultColumns] = useState(columnsProp ?? defaultAlertsTableColumns);
+    const [columns, setColumns] = useControllableState({
+      prop: columnsProp,
+      onChange: onColumnsChange,
+      defaultValue: applyColumnsConfiguration({
+        defaultColumns,
+        configuredColumns: configuration?.columns,
       }),
-      [columnsLocal, localStorageAlertsTableConfig, initialSort]
+    });
+    const [defaultVisibleColumns] = useState(visibleColumnsProp ?? columns.map((c) => c.id));
+    const [visibleColumns, setVisibleColumns] = useControllableState({
+      prop: visibleColumnsProp,
+      onChange: onVisibleColumnsChange,
+      defaultValue: configuration?.visibleColumns ?? defaultVisibleColumns,
+    });
+    const [sort, setSort] = useState<AlertsTableSortCombinations[]>(
+      configuration?.sort ?? initialSort
     );
-    const storageAlertsTable = useRef<AlertsTablePersistedConfiguration>(getStorageConfig());
 
-    storageAlertsTable.current = getStorageConfig();
+    // Persist table configuration on changes (using useUpdateEffect to avoid saving on first render)
+    useUpdateEffect(() => {
+      setConfiguration({
+        columns,
+        visibleColumns,
+        sort,
+      });
+    }, [columns, setConfiguration, sort, visibleColumns]);
 
-    const [sort, setSort] = useState<SortCombinations[]>(storageAlertsTable.current.sort);
+    const fieldsQuery = useFetchAlertsFieldsQuery(
```

Comment:
All the data fetching is done in this component but this single query was managed in the `useColumns` hook. I moved it here to simplify the flow and add it to the combined loading state computation. (I didn't invalidate this too in the refresh method since alert fields shouldn't change in the same session)